upload-api/tables/delegations.js

```diff
@@ -134,27 +151,54 @@ async function cidToDelegation (bucket, cid) {
   return delegation
 }
 
-/**
- * 
- * @param {import('../types').DelegationsBucket} bucket
- * @param {Ucanto.Delegation<Ucanto.Tuple<Ucanto.Capability>>[]} delegations
+/** 
+ * @param {import('../types').InvocationBucket} invocationBucket
+ * @param {import('../types').WorkflowBucket} workflowBucket
+ * @param {CID} invocationCid
+ * @param {CID} delegationCid
+ * @returns {Promise<Ucanto.Delegation>}
  */
-async function writeDelegations (bucket, delegations) {
-  return writeEntries(
-    bucket,
-    [...delegations].map((delegation) => {
-      const carBytes = delegationsToBytes([delegation])
-      const value = carBytes
-      return /** @type {[key: CID, value: Uint8Array]} */ ([delegation.cid, value])
-    })
+async function findDelegationInInvocation (invocationBucket, workflowBucket, invocationCid, delegationCid) {
+
+  // TODO is this right? I cargo culted some code from ucan-invocation.js and w3up/upload-api/src/access/delegate.js
+  // but am not highly confident it works
+
+  const invocationCidStr = invocationCid.toString()
+  const agentMessageWithInvocationCid = await invocationBucket.getInLink(
+    invocationCid.toString()
   )
-}
 
-/**
- * 
- * @param {import('../types').DelegationsBucket} bucket
- * @param {Iterable<readonly [key: CID, value: Uint8Array ]>} entries
- */
-async function writeEntries (bucket, entries) {
-  await Promise.all([...entries].map(([key, value]) => bucket.put(key, value)))
+  if (!agentMessageWithInvocationCid) {
+    throw new NoCarFoundForGivenReceiptError()
+  }
+
+  const agentMessageBytes = await workflowBucket.get(
+    agentMessageWithInvocationCid
+  )
+  if (!agentMessageBytes) {
+    throw new NoCarFoundForGivenReceiptError()
+  }
+
+  const agentMessage = await CAR.request.decode({
+    body: agentMessageBytes,
+    headers: {},
+  })
+  const invocation = agentMessage.invocations.find(
+    (inv) => inv.cid.toString() === invocationCidStr
+  )
+
+  if (!invocation) {
+    throw new NoInvocationFoundForGivenReceiptError()
+  }
+
+  const proofDelegations = invocation.proofs.flatMap((proof) =>
+    'capabilities' in proof ? [proof] : []
+  )
+  const foundDelegation = proofDelegations.find(proof => proof.cid)
+
+  if (!foundDelegation) {
+    throw new NoInvocationFoundForGivenReceiptError()
+  }
```

Comment:
this is the part I am less familiar with. The rest of the flow to get the CAR file where delegation is looks right